src/pages/studyView/StudyViewUtils.tsx

```diff
@@ -4006,3 +4065,533 @@ function ToastSuppressor(props: any) {
         </>
     );
 }
+
+export function getDefaultClinicalDataBinFilter(
+    attribute: ClinicalAttribute
+): ClinicalDataBinFilter & {
+    showNA?: boolean | undefined;
+} {
+    return {
+        attributeId: attribute.clinicalAttributeId,
+        disableLogScale: false,
+        showNA: true,
+    } as ClinicalDataBinFilter & { showNA?: boolean };
+}
+
+export function generateColorMapKey(id: string, value: string): string {
+    return `${id}.${value}`;
+}
+
+export async function invokeGenericAssayDataCount(
+    chartInfo: GenericAssayChart,
+    filters: StudyViewFilter
+) {
+    let result: GenericAssayDataCountItem[] = [];
```

Comment:
why set it to empty array and then overwrite?  
```
const result = await internalClient ...
```

Reply:
updated